src/Stratis.Bitcoin/Consensus/Rules/RuleContext.cs

```diff
@@ -67,25 +31,49 @@ public class RuleContext
         public bool SkipValidation { get; set; }
 
         /// <summary>The current tip of the chain that has been validated.</summary>
-        public ChainedHeader ConsensusTip { get; set; }
+        public ChainedHeader PreviousChainedHeader { get; set; }
```

Comment:
It is kind of confusing when something is current And its name is previous. Same with previous height below. Also comment previous height.

Reply:
BestChainedheader
Is better?

Reply:
Yes, but even the old name of consensus tip was fine.

Reply:
done